ingest/expression_files/expression_files.py

```diff
@@ -181,10 +178,85 @@ def insert(docs: List, collection_name: str, client):
                 f"Error caused by inserting into collection '{collection_name}': {e}"
             )
 
-    def load(self, gene_docs: List, data_array_docs: List):
+    def load(self, docs: List, collection_name: List):
         start_time = datetime.datetime.now()
-        GeneExpression.insert(gene_docs, self.COLLECTION_NAME, self.mongo_connection)
-        GeneExpression.insert(data_array_docs, "data_arrays", self.mongo_connection)
+        GeneExpression.insert(docs, collection_name, self.mongo_connection)
         self.info_logger.info(
-            f"Time to load {len(gene_docs) + len(data_array_docs)} models: {str(datetime.datetime.now() - start_time)}"
+            f"Time to load {len(docs)} models: {str(datetime.datetime.now() - start_time)}"
         )
+
+    def create_models(
+        self,
+        exp_cells: List,
+        exp_scores: List,
+        gene: str,
+        gene_id: str,
+        gene_models: List,
+        data_arrays: List,
+        num_processed: int,
+        force=False,
+    ):
+        """Creates models for a given gene and batches them for loading if
+            necessary.
+
+        After creating models, the amount of data arrays are checked to see if
+        models should be loaded into the database. Data arrays and gene models
+        will be loaded if the threshold, as specified in DATA_ARRAY_BATCH_SIZE,
+        is met.
+
+        Returns:
+            data_arrays: List[str]
+            gene_models: List[str]
+            num_processed: int
+                Amount of gene models created.
+            """
+        current_data_arrays = []
+        start_time = datetime.datetime.now()
+
+        model_id = ObjectId()
+        gene_models.append(
+            GeneExpression.create_gene_model(
+                name=gene,
+                study_file_id=self.study_file_id,
+                study_id=self.study_id,
+                gene_id=gene_id,
+                _id=model_id,
+            )
+        )
+        if len(data_arrays) > 0:
+            # Data arrays for cells
+            for data_array in GeneExpression.create_data_arrays(
+                name=f"{gene} Cells",
+                array_type="cells",
+                values=exp_cells,
+                linear_data_type="Gene",
+                linear_data_id=model_id,
+                **self.data_array_kwargs,
+            ):
+                current_data_arrays.append(data_array)
+            # Data arrays for expression values
+            for data_array in GeneExpression.create_data_arrays(
+                name=f"{gene} Expression",
+                array_type="expression",
+                values=exp_scores,
+                linear_data_type="Gene",
+                linear_data_id=model_id,
+                **self.data_array_kwargs,
+            ):
+                current_data_arrays.append(data_array)
```

Comment:
`data_array` is reused a few times to mean different things.  Using different variable names for each meaning would ease reading:

```suggestion
            for cell_data_array in GeneExpression.create_data_arrays(
                name=f"{gene} Cells",
                array_type="cells",
                values=exp_cells,
                linear_data_type="Gene",
                linear_data_id=model_id,
                **self.data_array_kwargs,
            ):
                current_data_arrays.append(cell_data_array)
            # Data arrays for expression values
            for exp_value_data_array in GeneExpression.create_data_arrays(
                name=f"{gene} Expression",
                array_type="expression",
                values=exp_scores,
                linear_data_type="Gene",
                linear_data_id=model_id,
                **self.data_array_kwargs,
            ):
                current_data_arrays.append(exp_value_data_array)
```

If those seem too verbose to you `cell_array` and `exp_value_array` would also work.